### PR TITLE
[Logs]: Update sender config to ignore empty logs_dd_url configuration coming from agent migrations

### PR DESCRIPTION
**Pull request by gaetan-deputier**
### What does this PR do?

Ignore empty logs_dd_url configuration
[Trello](https://trello.com/c/W8uHSqtr/940-agent-logsddurl-should-be-ignored-if-empty)

### Motivation

Some customers had issues setting up their agent for logs after migrating due to this bug as the empty logs_dd_url was making the agent fail

### CODE REVIEW

**Review by ajacquemot — DISMISSED** [review]

**Comment by gaetan-deputier** [comment]
Unit-tests from trace are flaky (seems an issue in the other opened Pull requests as well).

**Review by ajacquemot — DISMISSED** [review]

File: pkg/logs/sender/config.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ func BuildEndpoints() (*client.Endpoints, error) {
 		ProxyAddress: proxyAddress,
 	}
 	switch {
-	case config.Datadog.IsSet("logs_config.logs_dd_url"):
+	case config.Datadog.IsSet("logs_config.logs_dd_url") && config.Datadog.GetString("logs_config.logs_dd_url") != "":
```
Comment (ajacquemot) [inline pkg/logs/sender/config.go]: could we create a function that wraps this logic ?

Comment (gaetan-deputier) [reply pkg/logs/sender/config.go]: Done and unit test

File: pkg/logs/sender/config.go [hunk 2]
```diff
@@ -24,6 +24,10 @@ var logsEndpoints = map[string]int{
 	"agent-intake.logs.datad0g.eu":    443,
 }
 
+func IsSetAndNotEmpty(config config.Config, key string) bool {
```
Comment (ajacquemot) [inline pkg/logs/sender/config.go]: this method could be private, here you exposed it at package level.

Comment (gaetan-deputier) [reply pkg/logs/sender/config.go]: TIL about go encapsulation. Thanks for the comment, updated.

**Review by ajacquemot — APPROVED** [review]

**Review by achntrl — APPROVED** [review]

**Pull request merged**